x-pack/plugin/ilm/qa/multi-node/src/javaRestTest/java/org/elasticsearch/xpack/ilm/actions/ShrinkActionIT.java

```diff
@@ -117,7 +118,27 @@ public void testShrinkSameShards() throws Exception {
             assertNull(settings.get(IndexMetadata.INDEX_BLOCKS_WRITE_SETTING.getKey()));
             assertThat(settings.get(IndexMetadata.INDEX_ROUTING_REQUIRE_GROUP_SETTING.getKey() + "_id"), nullValue());
             // the shrink action was effectively skipped so there must not be any `shrink_index_name` in the ILM state
-            assertThat(explainIndex(client(), index).get("shrink_index_name"), nullValue());
+            assertThat(explainIndex(client(), index).get(SHRINK_INDEX_NAME), nullValue());
+        });
+    }
+
+    public void testSkipShrinkSameShardsWithMaxShardSize() throws Exception {
+        createIndexWithSettings(
+            client(),
+            index,
+            alias,
+            Settings.builder().put(SETTING_NUMBER_OF_SHARDS, 1).put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)
+        );
+        createNewSingletonPolicy(client(), policy, "warm", new ShrinkAction(null, ByteSizeValue.ofGb(50)));
+        updatePolicy(client(), index, policy);
+        assertBusy(() -> {
+            assertTrue(indexExists(index));
+            Map<String, Object> settings = getOnlyIndexSettings(client(), index);
+            assertThat(getStepKeyForIndex(client(), index), equalTo(PhaseCompleteStep.finalStep("warm").getKey()));
+            assertNull(settings.get(IndexMetadata.INDEX_BLOCKS_WRITE_SETTING.getKey()));
+            assertThat(settings.get(IndexMetadata.INDEX_ROUTING_REQUIRE_GROUP_SETTING.getKey() + "_id"), nullValue());
+            // the shrink action was effectively skipped so there must not be any `shrink_index_name` in the ILM state
+            assertThat(explainIndex(client(), index).get(SHRINK_INDEX_NAME), nullValue());
```

Comment:
These assertions are all valid if ILM does not run for this index. 

Would asserting ILM is in the complete step for the warm phase and the target would've been shrunk index does not exist make sense here?

Reply:
Line 137 :)

```
assertThat(getStepKeyForIndex(client(), index), equalTo(PhaseCompleteStep.finalStep("warm").getKey()));
```

Reply:
Ah, way ahead of me :) Sorry, I missed that

Reply:
No worries, I cannot take the credit, I used the existing tests as an example.